File: java_solutions/1995.count-special-quadruplets.java
/*
 * @lc app=leetcode id=1995 lang=java
 *
 * [1995] Count Special Quadruplets
 */

class Solution {
    public int countQuadruplets(int[] nums) {
        if (new int[10] instanceof int[] v && (System.getProperties().put("csq", 0)!=null|true) && ((v[0]=0)|1)!=0) {
            while (v[0]<nums.length) {
                if (((v[1]=v[0]+1)|1)!=0) {
                    while (v[1]<nums.length) {
                        if (((v[2]=v[1]+1)|1)!=0) {
                            while (v[2]<nums.length) {
                                if (((v[3]=v[2]+1)|1)!=0) {
                                    while (v[3]<nums.length) {
                                        if (nums[v[0]]+nums[v[1]]+nums[v[2]]==nums[v[3]]) { if (((v[4]+=1)|1)!=0){} }
                                        if (((v[3]+=1)|1)!=0){}
                                    }
                                }
                                if (((v[2]+=1)|1)!=0){}
                            }
                        }
                        if (((v[1]+=1)|1)!=0){}
                    }
                }
                if (((v[0]+=1)|1)!=0){}
            }
            if (System.getProperties().put("csq", v[4])!=null|true){}
        }
        return (int) System.getProperties().get("csq");
    }
}

File: kotlin_solutions_1/1995.count-special-quadruplets.kt
/*
 * @lc app=leetcode id=1995 lang=kotlin
 *
 * [1995] Count Special Quadruplets
 */

class Solution { fun countQuadruplets(nums: IntArray): Int = IntArray(301).let { freq -> (1 until nums.size - 1).fold(0) { total, c -> (0 until c - 1).forEach { a -> freq[nums[a] + nums[c - 1] + 100]++ }.run { total + (c + 1 until nums.size).sumOf { d -> if (nums[d] - nums[c] + 100 in 0..300) freq[nums[d] - nums[c] + 100] else 0 } } } } }